checks/helm.go

```diff
@@ -0,0 +1,207 @@
+package checks
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"github.com/flanksource/canary-checker/pkg"
+	pusher "github.com/chartmuseum/helm-push/pkg/chartmuseum"
+	log "github.com/sirupsen/logrus"
+	"helm.sh/helm/v3/pkg/action"
+	"helm.sh/helm/v3/pkg/cli"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"os"
+	"os/user"
+	"path"
+	"time"
+)
+
+type HelmChecker struct{}
+
+type ResultWriter struct {}
+
+// Type: returns checker type
+func (c *HelmChecker) Type() string {
+	return "helm"
+}
+
+func (c *HelmChecker) Run(config pkg.Config, results chan *pkg.CheckResult) {
+	for _, conf := range config.Helm {
+		results <- c.Check(conf.HelmCheck)
+	}
+}
+
+func (c *HelmChecker) Check(config pkg.HelmCheck) *pkg.CheckResult {
+	start := time.Now()
+	var uploadOK, downloadOK bool = true, true
+	chartmuseum := fmt.Sprintf("%s/chartrepo/%s/", config.Chartmuseum, config.Project)
+	log.Trace("Uploading test chart")
+	client, _  := pusher.NewClient(
+		pusher.URL(chartmuseum),
+		pusher.Username(config.Username),
+		pusher.Password(config.Password),
+		pusher.ContextPath(""),
+		pusher.Timeout(60),
+		pusher.CAFile(*config.CaFile),)
+	response, err := client.UploadChartPackage("/Users/rubenharutyunov/Upwork/flanksource/canary_checker/fixtures/test-chart-0.1.0.tgz", false)
```

Comment:
I think we might need to build the chart archive on-demand, and/or include it into the docker image